woocommerce/woocommerce-ios
# [Tablet Orders] Update scan button rendering based on size class

Closes: #11747


## Description
This PR addresses how we render the "scan" button from Order creation based on size class: 

The existing design shows a scanner button in the Order Summary screen, either in the "Add Products" row when no products have been added to the order yet, or in the component that holds products:
| Existing design: Order has no products | Existing design: Order has products |
|--------|--------|
| <img width="206" alt="Screenshot 2024-02-15 at 14 58 54" src="https://github.com/woocommerce/woocommerce-ios/assets/3812076/31bb70b9-a508-4dfe-9e92-c4cbcb0835aa"> | <img width="219" alt="Screenshot 2024-02-15 at 14 59 07" src="https://github.com/woocommerce/woocommerce-ios/assets/3812076/d53f4b6e-5c6f-4aa4-b564-ba0d7cf46e90"> | 

The proposed design for tablets changes this by removing the scan button from the "Order Summary" view, and adding it to the "Product Selector" view instead, to the right side of the search bar:
| Proposed changes |
|--------|
| <img width="579" alt="Screenshot 2024-02-15 at 14 58 02" src="https://github.com/woocommerce/woocommerce-ios/assets/3812076/2e5187ec-82e5-4104-89a7-96e6e68896a7"> |

## Testing instructions
* Keep the `sideBySideViewForOrderForm` feature flag set to `false`
* On an iPhone that does not support split views (eg: iPhone 15), go to Orders > Tap `+` > Observe that the scanner button appears as always, both when no products are added to the order, as well as when products are added. Confirm both in portrait and landscape mode.
* Enable the `sideBySideViewForOrderForm` feature flag set to `true`
* Repeat the previous step. You should see the same results.
* On a tablet, go to Orders > Tap `+` > Observe that the scanner shows in the product selector, near to search products, and does not appear in the order summary.
* Tapping the button will not trigger the scanning flow, this will be tackled separately on #12011

## Caveats
* As mentioned in the testing instructions, tapping the button will not trigger the scanning flow, this will be tackled separately on #12011.
* The polished design of the header has not been tackled yet, this will be done on #11748, this PR only deals with hiding and showing the scan button when appropriate based on size class.

_Side-note: I found useful to test as well on an iPhone that supports split views to see if the render works correctly (like the 15 Plus) while the design is not the best yet due to the small screen and there are some bugs, the portrait uses ModalOnModal while landscape uses SideBySide._
